api/preprints/serializers.py

```diff
@@ -306,3 +313,71 @@ class PreprintCitationSerializer(NodeCitationSerializer):
 
     class Meta:
         type_ = 'preprint-citation'
+
+
+class PreprintContributorsSerializer(NodeContributorsSerializer):
+    """ Separate from UserSerializer due to necessity to override almost every field as read only
+    """
+    preprint = RelationshipField(
+        related_view='preprints:preprint-detail',
+        related_view_kwargs={'preprint_id': '<preprint._id>'}
+    )
+
+    node = HideIfPreprint(RelationshipField(
+        related_view='nodes:node-detail',
+        related_view_kwargs={'node_id': '<node._id>'}
+    ))
+
+    class Meta:
+        type_ = 'contributors'
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse(
+            'preprints:preprint-contributor-detail',
+            kwargs={
+                'user_id': obj.user._id,
+                'preprint_id': self.context['request'].parser_context['kwargs']['preprint_id'],
+                'version': self.context['request'].parser_context['kwargs']['version']
+            }
+        )
+
+
+class PreprintContributorsCreateSerializer(NodeContributorsCreateSerializer, PreprintContributorsSerializer):
+    """
+    Overrides PreprintContributorsSerializer to add email, full_name, send_email, and non-required index and users field.
+
+    id and index redefined because of the two serializers we've inherited
+    """
+    id = IDField(source='_id', required=False, allow_null=True)
+    index = ser.IntegerField(required=False, source='_order')
+
+    email_preferences = ['preprint', 'false']
+
+    def get_proposed_permissions(self, validated_data):
+        return validated_data.get('permission') or osf_permissions.WRITE
+
+
+class PreprintContributorDetailSerializer(NodeContributorDetailSerializer, PreprintContributorsSerializer):
+    """
+    Overrides NodeContributorDetailSerializer to set the preprint instead of the node
+
+    id and index redefined because of the two serializers we've inherited
+    """
+    id = IDField(required=True, source='_id')
+    index = ser.IntegerField(required=False, read_only=False, source='_order')
+
+
+class PreprintProviderSerializer(NodeProviderSerializer):
```

Comment:
Drive-by: this name seems overloaded

Reply:
you're right, this is confusing. Was just making this a parallel to the NodeProvider serializer, but will rename.